File: wlr/OGUR/OGUR/Strategies/FleeStrategy.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OGUR.Creatures;
using OGUR.Collision;

namespace OGUR.Strategies
{
    class FleeStrategy:IStrategy
    {
        public FleeStrategy(ICreature parent, params int[] targetTypes)
            : base(parent)
        {
            foreach (var targetType in targetTypes)
            {
                m_targets.AddTargetTypes(targetType);
            }
        }

        private Point2 _transfer = new Point2(0, 0);
        public override void Act()
        {
            if (AbleToMove())
            {
                _transfer = targetPath.GetNextMove();
                if(_transfer != null)
                {
                    nextMove.Copy(m_parent.GetLocation().Add(_transfer.Minus(m_parent.GetLocation()).Rotate180()));
                    m_parent.MoveTo(nextMove);
                }
            }
        }
    }
}

File: proj/src/aigilas/strategies/impl/FleeStrategy.java
package aigilas.strategies.impl;

import aigilas.creatures.ICreature;
import aigilas.strategies.IStrategy;
import aigilas.strategies.Strategy;
import spx.bridge.ActorType;
import spx.core.Point2;

public class FleeStrategy extends IStrategy {
    public FleeStrategy(ICreature parent, ActorType... targetTypes)

    {
        super(parent, Strategy.Flee);

        for (ActorType targetType : targetTypes) {
            _targets.addTargetTypes(targetType);
        }
    }

    private Point2 _transfer = new Point2(0, 0);

    @Override
    public void act() {
        if (AbleToMove()) {
            _transfer = targetPath.getNextMove();
            if (_transfer != null) {
                nextMove.copy(_parent.getLocation().add(_transfer.minus(_parent.getLocation()).rotate180()));
                _parent.moveTo(nextMove);
            }
        }
    }
}
